src/api/l_headset.c

```diff
@@ -578,3 +587,93 @@ int luaopen_lovr_headset(lua_State* L) {
   headsetRenderData.ref = LUA_NOREF;
   return 1;
 }
+
+// TEMPORARY HANDS LIBRARY
+
+#include "headset/oculus_mobile_bridge.h"
+#include "headset/oculus_mobile.h"
+
+static void pushTableOfNumbers(lua_State* L, int idx, int count, ...) { // Takes floats
+  va_list args;
+  va_start(args, count);
+
+  lua_newtable(L);
+  for(int c = 0; c < count; c++) {
+    // Note we pass floats but they become doubles because of some garbage about how va_list works.
+    // This is probably okay becuase lua_number is probably a double anyway.
+    lua_pushnumber(L, va_arg(args, double));
+    lua_rawseti (L, -2, c+1);
+  }
+  lua_rawseti (L, -2, idx);
+
+  va_end(args);
+}
+
+static int l_lovrHeadsetHandsIsHand(lua_State* L) {
+  Device device = luax_optdevice(L, 1);
+  lovrAssert(device == DEVICE_HAND_LEFT || device == DEVICE_HAND_RIGHT, "Only works with hands");
+  LovrOculusMobileHands *hand = &lovrOculusMobileHands[device == DEVICE_HAND_RIGHT];
+
+  int points = 0;
+  lua_pushboolean(L, hand->live);
+
+  return 1;
+}
+
+static int l_lovrHeadsetHandsGetPoints(lua_State* L) {
+  Device device = luax_optdevice(L, 1);
+  lovrAssert(device == DEVICE_HAND_LEFT || device == DEVICE_HAND_RIGHT, "Only works with hands");
+  LovrOculusMobileHands *hand = &lovrOculusMobileHands[device == DEVICE_HAND_RIGHT];
+
+  int points = 0;
+  lua_newtable(L);
+  for(int c = 0; c < hand->handPoses.members; c++) {
+    BridgeLovrPose *pose = &hand->handPoses.poses[c];
+    lua_newtable(L);
+    pushTableOfNumbers(L, 1, 3, pose->x, pose->y, pose->z);
+    pushTableOfNumbers(L, 2, 4, pose->q[0], pose->q[1], pose->q[2], pose->q[3]);
+    lua_rawseti (L, -2, ++points);
+  }
+
+  return 1;
+}
+
+static int l_lovrHeadsetHandsGetBoneNames(lua_State* L) {
+  Device device = luax_optdevice(L, 1);
+  lovrAssert(device == DEVICE_HAND_LEFT || device == DEVICE_HAND_RIGHT, "Only works with hands");
+  LovrOculusMobileHands *hand = &lovrOculusMobileHands[device == DEVICE_HAND_RIGHT];
+
+  lua_newtable(L);
+
+  for(int c = 0; c < hand->bones.members; c++) {
+    lua_pushstring(L, hand->bones.strings[c]);
+    lua_rawseti (L, -2, c+1);
+  }
+
+  return 1;
+}
+
+static int l_lovrHeadsetHandsGetConfidence(lua_State* L) {
```

Comment:
This returns a table where indices 1-5 contain the confidence values for the 5 fingers, and index 0 contains the confidence value for the entire hand. A confidence value is a float between 0 and 1. In the current implementation it is always EXACTLY either 0 or 1 because the current vrapi only gives us an enum containing "low value" or "high value". I have suspicions that might change but who knows.